src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -239,27 +249,260 @@ private static void getUnpublishedCourseList(String canvasTermIdForSisTermId, Co
-						.setStartDate((String)course.get("start_at"))
-						.setEndDate((String)course.get("end_at"));
+						.setStartDate(Utils.changeToDate((String)course.get("start_at")))
+						.setEndDate(Utils.changeToDate((String)course.get("end_at")));
 
 				coursesForDelete.addCourse(aCourse);
+				checkForEndDateContentActivityInACourse(aCourse, apiHandler,coursesForDelete);
 			}
 			String nextPageUrl = getNextPageUrl(httpResponse);
 			if(nextPageUrl!=null) {
-				getUnpublishedCourseList(canvasTermIdForSisTermId,coursesForDelete,apiHandler,nextPageUrl);
+				manageCoursesDeletion(previousTerm,coursesForDelete,apiHandler,nextPageUrl);
 			}
 		} catch (JsonParseException e1) {
-			M_log.error("JsonParseException occured getUnpublishedCourseList() : ",e1);
+			M_log.error("JsonParseException occured manageCoursesDeletion() : ",e1);
 		} catch(JsonMappingException e1) {
-			M_log.error("JsonMappingException occured getUnpublishedCourseList() : ",e1);
+			M_log.error("JsonMappingException occured manageCoursesDeletion() : ",e1);
 		}  catch (IOException e) {
-			M_log.error("IOException occured getUnpublishedCourseList( ):" ,e);
+			M_log.error("IOException occured manageCoursesDeletion( ):" ,e);
+		}
+	}
+	/*
+	 * For deletion of a course we need to check  1) courseEndDate with respect to current system date.If endDate is past the current system
+	 * date then course becomes eligible to for delete 2) after passing the endDate check we will be checking if the course has any content in it. We will be querying 
+	 * various canvas tool api's  to determine.3)If their is no content exist in the course check should be made for activity. if all these check show the course never been used 
+	 * then we will delete the course
+	 */
+	private static void checkForEndDateContentActivityInACourse(Course course,ApiCallHandler apiHandler,CoursesForDelete coursesForDelete) {
+		if(!isCourseEndDateIsPastTheCurrentDate(course)) {
+			if(!isTheirContentInCourse(course,apiHandler)) {
+				if(!isTheirActivityInCourse(course,apiHandler,null)) {
+					deleteTheCourse(course,apiHandler,coursesForDelete);
+				}
+			}
+		}
+		
+	}
+    /*
+     * Some time a course End date may be set after the term end date, in that cases we don't want to delete the course.
+     * Course end date can be null as well i.e it might not be set at all.
+     */
+	private static boolean isCourseEndDateIsPastTheCurrentDate(Course course) {
+		M_log.debug("isCourseEndDateIsPastTheCurrentDate: "+course.getCourseId()+" and endDate: "+course.getEndDate());
+		if(course.getEndDate()==null || course.getEndDate().isBefore(new DateTime(DateTimeZone.UTC))) {
+			M_log.debug("!!! Course eligible for Delete !!!" );
+			return false;
+		}
+		return true;
+	}
+
+	private static boolean isTheirContentInCourse(Course course, ApiCallHandler apiHandler) {
+		M_log.debug("isTheirContentInCourse: "+course.getCourseId());
+		boolean content=false;
+		if(areTheirFiles(course.getCourseId(),apiHandler)) {
```

Comment:
I will refactor if/else logic. I feel coding for logic to decide which test make more true would be really complicated. instead i can just change the logging to info from debug. I think that requirement is for us and while testing the utility we can find out that easily and change it as needed. The order of the list is not very off than how the canvas tools are used in real life.